plugin/src/main/java/org/autorefactor/refactoring/rules/AndroidViewHolderRefactoring.java

```diff
@@ -0,0 +1,320 @@
+
+                    }
+                    r.remove(visitor.viewAssignmentStatement);
+                    return DO_NOT_VISIT_SUBTREE;
+                }
+            }
+        }
+        return VISIT_SUBTREE;
+    }
+
+    private static boolean isInflateMethod(MethodInvocation node) {
+        return isMethod(node, "android.view.LayoutInflater", "inflate", "int", "android.view.ViewGroup")
+                || isMethod(node, "android.view.LayoutInflater", "inflate", "int", "android.view.ViewGroup", "boolean")
+                || isMethod(node, "android.view.LayoutInflater", "inflate", "org.xmlpull.v1.XmlPullParser",
+                        "android.view.ViewGroup")
+                || isMethod(node, "android.view.LayoutInflater", "inflate", "org.xmlpull.v1.XmlPullParser",
+                        "android.view.ViewGroup", "boolean");
+    }
+
+    private class GetViewVisitor extends ASTVisitor {
+        private SimpleName viewVariable = null;
+        private Statement viewAssignmentStatement;
+        private VariableDeclarationFragment viewVariableDeclarationFragment = null;
+        private Assignment viewVariableAssignment = null;
+        private ReturnStatement returnStatement = null;
+
+        GetViewVisitor() {
+        }
+
+        @Override
+        public boolean visit(MethodInvocation node) {
+            if (isInflateMethod(node)) {
+                this.viewVariableDeclarationFragment = (VariableDeclarationFragment) getFirstAncestorOrNull(node,
+                        VariableDeclarationFragment.class);
+                if (viewVariableDeclarationFragment != null) {
+                    this.viewVariable = viewVariableDeclarationFragment.getName();
+                    this.viewAssignmentStatement = getAncestor(viewVariableDeclarationFragment,
+                            VariableDeclarationStatement.class);
+                } else {
+                    this.viewVariableAssignment = (Assignment) getFirstAncestorOrNull(node, Assignment.class);
+                    if (viewVariableAssignment != null) {
+                        this.viewVariable = (SimpleName) viewVariableAssignment.getLeftHandSide();
+                        this.viewAssignmentStatement = getAncestor(viewVariableAssignment, ExpressionStatement.class);
+                    }
+                }
+                return DO_NOT_VISIT_SUBTREE;
+            }
+            return VISIT_SUBTREE;
+        }
+
+        public boolean visit(ReturnStatement node) {
+            this.returnStatement = node;
+            return VISIT_SUBTREE;
+        }
+
+        public boolean isInflateInsideIf() {
+            if (this.viewAssignmentStatement != null) {
+                if (getFirstAncestorOrNull(this.viewAssignmentStatement, IfStatement.class) != null) {
+                    return true;
+                } else if (getFirstAncestorOrNull(this.viewAssignmentStatement, SwitchStatement.class) != null) {
```

Comment:
Everywhere in this file please use `getAncestor()` instead of `getFirstAncestorOrNull()` where possible.
`getFirstAncestorOrNull()` is designed for multiple values types.